src/main/resources/org/jenkinsci/plugins/credentialsbinding/impl/BindingStep/config.jelly

```diff
@@ -25,6 +25,7 @@ THE SOFTWARE.
 
 <?jelly escape-by-default='true'?>
 <j:jelly xmlns:j="jelly:core" xmlns:f="/lib/form">
+    <j:set var="expressionAllowed" value="false"/>
```

Comment:
Both, really.

First, passing `true` causes an NPE, which ultimately is due to the fact that the POSTed JSON looks like

```json
"credentialsId":["mycreds",""]
```

rather than the expected

```json
"credentialsId":"mycreds"
```

which indicates that `buildFormTree` is choking on the way `select.jelly` is written. Frankly I am not sure how it works for the freestyle `SecretBuildWrapper`; somehow the correct JSON is sent in that case.

Second, for `BindingStep`, the configuration form is only displayed in _Snippet Generator_ for a Pipeline job. In this case there is no advantage to allowing an expression, and indeed it is just misleading and confusing. _Any_ parameter to a step which you want to make dynamic in some way you can do so using normal Groovy code; there is no reason for this one to be treated differently.